### PR TITLE
v8 Upgrade Handler

### PR DESCRIPTION
**Pull request by sampocs**
Closes: #XXX

## Context and purpose of the change
Upgrade handler for v8

## Brief Changelog
* Adds upgrade handler for v8
* Adds evmos airdrop and resets existing airdrops

## Author's Checklist

I have...

- [ ] Run and PASSED locally all GAIA integration tests
- [ ] If the change is contentful, I either:
    - [ ] Added a new unit test OR 
    - [ ] Added test cases to existing unit tests
- [ ] OR this change is a trivial rework / code cleanup without any test coverage

If skipped any of the tests above, explain.

## Reviewers Checklist

*All items are required. Please add a note if the item is not applicable and please add
your handle next to the items reviewed if you only reviewed selected items.*

I have...

- [ ] reviewed state machine logic
- [ ] reviewed API design and naming
- [ ] manually tested (if applicable)
- [ ] confirmed the author wrote unit tests for new logic
- [ ] reviewed documentation exists and is accurate

## Documentation and Release Note

  - [ ] Does this pull request introduce a new feature or user-facing behavior changes? 
  - [ ] Is a relevant changelog entry added to the `Unreleased` section in `CHANGELOG.md`?
  - [ ] This pull request updates existing proto field values (and require a backend and frontend migration)? 
  - [ ] Does this pull request change existing proto field names (and require a frontend migration)?
  How is the feature or change documented? 
      - [ ] not applicable
      - [ ] jira ticket `XXX` 
      - [ ] specification (`x/<module>/spec/`) 
      - [ ] README.md 
      - [ ] not documented

### CODE REVIEW

**Review by sampocs — COMMENTED** [review]
Reminder we need to add distributor address for evmos

File: app/upgrades/v8/upgrades.go [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+		ctx.Logger().Info("Resetting airdrop claims...")
+		for _, claimType := range ResetAirdropIdentifiers {
+			if err := claimKeeper.ResetClaimStatus(ctx, claimType); err != nil {
+				return vm, errorsmod.Wrapf(err, fmt.Sprintf("unable to reset %s claim status", claimType))
+			}
+		}
+
+		// Add the evmos airdrop
+		ctx.Logger().Info("Adding evmos airdrop...")
+		blockTime := uint64(ctx.BlockTime().Unix())
```
Comment (shellvish) [inline app/upgrades/v8/upgrades.go:50]: we should pick a blocktime in the future I think

File: app/upgrades/v8/upgrades_test.go [hunk 2]
```diff
@@ -0,0 +1,139 @@
...
+	osmoAirdrop := claimParams.Airdrops[0]
+	evmosAirdrop := claimParams.Airdrops[1]
+
+	// Check that the params of the osmo airdrop were reset
+	s.Require().Equal(osmoAirdropId, osmoAirdrop.AirdropIdentifier, "osmo airdrop identifier")
+	s.Require().Zero(osmoAirdrop.ClaimedSoFar.Int64(), "osmo claimed so far")
+
+	// Check that the params of the evmos airdrop were initialized
+	s.Require().Equal(v8.EvmosAirdropIdentifier, evmosAirdrop.AirdropIdentifier, "evmos airdrop identifier")
+	s.Require().Zero(evmosAirdrop.ClaimedSoFar.Int64(), "cevmos laimed so far")
```
Comment (shellvish) [inline app/upgrades/v8/upgrades_test.go:112]: nit, "evmos claimed so far"

**Review by shellvish — APPROVED** [review]
Looks good to me! 

Only outstanding question, is there something we can do in the upgrade handler to print out logs when the Evmos airdrop is being added, and when the claims are being reset? Maybe something like "if index % 500 == 0, print index" ?

File: app/upgrades/v8/upgrades.go (hunk 1 above)
Comment (sampocs) [reply app/upgrades/v8/upgrades.go:50]: good call! should we shoot for next monday?

File: app/upgrades/v8/upgrades_test.go (hunk 2 above)
Comment (sampocs) [reply app/upgrades/v8/upgrades_test.go:112]: good catch!

**Pull request merged**